tokio-quiche/src/socket/capabilities.rs

```diff
@@ -116,6 +117,32 @@ impl SetSockOpt for Ipv6MtuDiscoverProbe {
     }
 }
 
+#[cfg(target_os = "linux")]
+#[derive(Clone)]
+struct RcvMark;
+
+#[cfg(target_os = "linux")]
+impl SetSockOpt for RcvMark {
+    type Val = ();
+
+    fn set<F: AsFd>(&self, fd: &F, _val: &Self::Val) -> nix::Result<()> {
+        let ret = unsafe {
+            libc::setsockopt(
+                fd.as_fd().as_raw_fd(),
+                SOL_SOCKET,
+                SO_RCVMARK,
+                &1 as *const c_int as *const c_void,
+                std::mem::size_of::<c_int>() as socklen_t,
+            )
+        };
+
+        match ret {
+            0 => Ok(()),
+            _ => Err(Errno::last()),
+        }
+    }
+}
+
```

Comment:
- Can you add documentation to this and link the Linux man page (couldnt actually find the linux docs on this). e.g. why setting value to `1` is appropriate. 
- Can we pull out `1` as a named const